enclave/core/sgx/reloc.c

```diff
@@ -40,7 +40,10 @@ bool oe_apply_relocations(void)
         /* Relocate the reference */
         if (reloc_type == R_X86_64_RELATIVE)
         {
-            *dest = (uint64_t)(baseaddr + p->r_addend);
+            int64_t addend = p->r_addend;
```

Comment:
FYI, one relocation type that I encountered when trying out many static libraries was
`R_X86_64_NONE`. This is a noop reloc. The linker was able to optimize away the relocation, but not the entry itself.
We need to implement this only if the final version of FIPS crypto has such a relocation.